lib/matplotlib/backends/backend_pgf.py

```diff
@@ -836,16 +835,10 @@ def print_pgf(self, fname_or_fh, *args, **kwargs):
         if kwargs.get("dryrun", False):
             self._print_pgf_to_fh(None, *args, **kwargs)
             return
-
-        # figure out where the pgf is to be written to
-        if isinstance(fname_or_fh, str):
-            with open(fname_or_fh, "w", encoding="utf-8") as fh:
-                self._print_pgf_to_fh(fh, *args, **kwargs)
-        elif is_writable_file_like(fname_or_fh):
-            fh = codecs.getwriter("utf-8")(fname_or_fh)
-            self._print_pgf_to_fh(fh, *args, **kwargs)
-        else:
-            raise ValueError("filename must be a path")
+        with cbook.open_file_cm(fname_or_fh, "w", encoding="utf-8") as file:
+            if not cbook.file_requires_unicode(file):
+                file = codecs.getwriter("utf-8")(file)
```

Comment:
Can we not clobber `file` like this, please?

Reply:
Also, when reassigning a variable within a context like this, I don't think the close() gets called on the new object because the context manager is holding a reference to the original object.

Reply:
Clobbering what?  `file` is not a builtin (... anymore).  And I'd say it's a much nicer variable name than `fh`...

The fact that the codec writer doesn't get closed doesn't matter (and anyways it will get closed by the GC), what matters is that the file itself gets closed.
```
import codecs
import contextlib


class filelike:  # mocks file
    def close(self):
        print("closed")


@contextlib.contextmanager
def cm():  # mocks open_file_cm
    print("enter")
    f = filelike()
    yield f
    f.close()
    print("exit")


with cm() as file:
    file = codecs.getwriter("utf-8")(file)


print("done")
```
prints
```
enter
closed
exit
done
```
so the file is indeed closed before the contextmanager is exited.